src/simulation/simulation.jl

```diff
@@ -266,6 +266,62 @@ function _check_folder(sim::Simulation)
     end
 end
 
+# Compare initial conditions for all `InitialConditionType`s with the
+# `requires_reconciliation` trait across `models`, log @info messages for mismatches
+function _initial_conditions_reconciliation!(
+    models::Vector{DecisionModel{GenericOpProblem}})
+    model_names = get_name.(models)
+    has_mismatches = false
+    @info "Reconciling initial conditions across models $(join(model_names, ", "))"
+    # all_ic_keys: all the `ICKey`s that appear in any of the models
+    # TODO: incorporate requires_reconciliation
+    all_ic_keys = union(keys.(get_initial_conditions.(models))...)
+    # all_ic_values: Dict{ICKey, Dict{model_index, Dict{component_name, ic_value}}}
+    all_ic_values = Dict()
+    for ic_key in all_ic_keys
+        if !requires_reconciliation(get_entry_type(ic_key))
+            @debug "Skipping initial conditions reconciliation for $(get_entry_type(ic_key)) due to false requires_reconciliation"
+            continue
+        end
+        # ic_vals_per_model: Dict{model_index, Dict{component_name, ic_value}}
+        ic_vals_per_model = Dict()
+        for (i, model) in enumerate(models)
+            ics = PSI.get_initial_conditions(model)
+            haskey(ics, ic_key) || continue
+            # ic_vals_per_component: Dict{component_name, ic_value}
+            ic_vals_per_component =
+                Dict(get_name(get_component(ic)) => get_condition(ic) for ic in ics[ic_key])
+            ic_vals_per_model[i] = ic_vals_per_component
+        end
+
+        # Assert that all models have the same components for current ic_key
+        @assert allequal(Set.(keys.(values(ic_vals_per_model)))) "For IC key $ic_key, not all models have the same components"
+
+        # For each component in current ic_key, compare values across models
+        component_names = collect(keys(first(values(ic_vals_per_model))))
```

Comment:
there is no need to call collect here to loop over the keys.

Reply:
Removed